packages/web3-core-method/lib/methods/AbstractMethod.js

```diff
@@ -23,21 +23,21 @@
 import isFunction from 'lodash/isFunction';
 import isString from 'lodash/isString';
 import cloneDeep from 'lodash/cloneDeep';
-import {PromiEvent} from 'web3-core-promievent';
 
 export default class AbstractMethod {
     /**
      * @param {String} rpcMethod
      * @param {Number} parametersAmount
      * @param {Utils} utils
      * @param {Object} formatters
+     * @param {AbstractWeb3Module} moduleInstance
      *
      * @constructor
      */
-    constructor(rpcMethod, parametersAmount, utils, formatters) {
+    constructor(rpcMethod, parametersAmount, utils, formatters, moduleInstance) {
         this.utils = utils;
         this.formatters = formatters;
-        this.promiEvent = new PromiEvent();
+        this.moduleInstance = moduleInstance;
```

Comment:
Perhaps it's not my place to ask, but wouldn't this add a copy of the module to all methods? I haven't tested this branch, but instead of copying the whole module instance, perhaps modules could have a concrete `moduleInstance.config`? (I don't remember the full structure of the package, but this might also avoid circular references somewhere unexpected.)

 Maybe this is preemptive optimization, but I thought I'd put it out there.

Reply:
The garbage collector of JS will collect and remove the ``AbstractMethod`` after the execution of the method. The ``AbstractMethod`` classes will be created in the ``MethodProxy`` when a declared method gets called. It will not copy the ``moduleInstance`` each time. It will just set the reference of the pointer. This means ``this.moduleInstance`` is just a "symlink" to the actual object.

Reply:
I see. I wasn't sure if the `moduleInstance` would end up as a pointer or as a copy. It only takes one random line referencing the object to end up allocated till unmounting.

Reply:
<img width="434" alt="Bildschirmfoto 2019-03-11 um 22 05 04" src="https://user-images.githubusercontent.com/8671781/54158940-072b6b00-444c-11e9-812e-a55b2723df10.png">

Maybe it works directly with the ES6 structure, but the `.umd.js` files don't seem to be correctly collected. Maybe for this release working on the `.umd.js` transpiler settings would be beneficial. The image above is taken from the production build of the `getBlock` example. https://codesandbox.io/s/0ykmkv2380

Reply:
This is the actual production build heap snapshot after some digging. I've tested the `.umd.js` every now and then and its behavior strays quite a bit from its `.esm.js` counterpart.

<img width="474" alt="Bildschirmfoto 2019-03-11 um 22 43 14" src="https://user-images.githubusercontent.com/8671781/54160261-5b841a00-444f-11e9-9732-52e767b53a31.png">

Reply:
There are still some points I could optimize the memory usage and other stuff. But I will do this later because I've already broken the memory usage down from ~550mb to ~280mb. Thanks for your tests! :-)

Reply:
Sorry if I came across as pedantic. What worries me overall is that the
transpiling to browser JS might end up having unexpected behavior. Every
now and then it happens to me that the ES6 or CJS code works but the
browser code doesn't. Probably not a pressing issue, but something I'd
avoid neglecting since the package is widely used client-side, but the
tests don't strictly cover it.